src/aoptk/relationships/zero_shot_classification.py

```diff
@@ -0,0 +1,77 @@
+from typing import TYPE_CHECKING
+from transformers import pipeline
+from aoptk.chemical import Chemical
+from aoptk.effect import Effect
+from aoptk.relationships.find_relationship import FindRelationships
+from aoptk.relationships.relationship import Relationship
+
+if TYPE_CHECKING:
+    from aoptk.chemical import Chemical
+    from aoptk.effect import Effect
+
+
+class ZeroShotClassification(FindRelationships):
+    """Zero-shot classification for finding relationships between chemicals and effects in text."""
+
+    task = "zero-shot-classification"
+    threshold = 0.6
+    margin = 0.15
+
+    def __init__(
+        self,
+        text: str,
+        chemicals: list[Chemical],
+        effects: list[Effect],
+        model: str = "facebook/bart-large-mnli",
+    ):
+        self.text = text
+        self.chemicals = chemicals
+        self.effects = effects
+        self.model = model
+        self.classifier = pipeline(self.task, model)
+
+    def find_relationships(self) -> list[Relationship]:
+        """Find relationships between chemicals and effects using zero-shot classification."""
+        relationships = []
+        for chemical, effect in product(self.chemicals, self.effects):
+            if relationship := self._classify_relationship(chemical, effect):
+                relationships.append(relationship)
+        return relationships
+
+    def _classify_relationship(self, chemical: Chemical, effect: Effect) -> Relationship | None:
+        """Classify the relationship between a chemical and an effect."""
+        candidate_labels = [
+            f"{chemical} induces {effect}",
+            f"{chemical} does not induce {effect}",
+            f"{chemical} prevents or does not prevent {effect}",
+            f"{chemical} has no known association with {effect}",
+        ]
+
+        result = self.classifier(self.text, candidate_labels)
+
+        labels = result["labels"]
+        scores = result["scores"]
+
+        top_label = labels[0]
+        top_score = scores[0]
+        second_score = scores[1]
+
+        if self._is_prediction_confident_enough(top_score, second_score) and (
+            relationship_type := self._select_relationship_type(top_label, candidate_labels)
+        ):
+            return Relationship(relationship=relationship_type, chemical=chemical, effect=effect)
+        return None
+
+    def _is_prediction_confident_enough(self, top_score: int, second_score: int) -> bool:
+        """Check if the prediction is confident enough based on threshold and margin."""
+        return top_score >= self.threshold and (top_score - second_score) >= self.margin
+
+    def _select_relationship_type(self, top_label: str, candidate_labels: list[str]) -> str | None:
+        """Select the relationship type based on the top label."""
+        if top_label == candidate_labels[0]:
+            return "positive"
+        if top_label == candidate_labels[1]:
+            return "negative"
+        return None
```

Comment:
If we want the relationships to be user defined, then we would have to also ask the user for this. How is this currently used? Is whether a relationship is positive or negative somehow presented in the output?